distribution/tools/launchers/src/main/java/org/elasticsearch/tools/launchers/DefaultSystemMemoryInfo.java

```diff
@@ -10,32 +10,29 @@
 
 import com.sun.management.OperatingSystemMXBean;
 
-import org.elasticsearch.tools.java_version_checker.JavaVersion;
 import org.elasticsearch.tools.java_version_checker.SuppressForbidden;
 
 import java.lang.management.ManagementFactory;
+import java.util.function.LongSupplier;
 
 /**
  * A {@link SystemMemoryInfo} which delegates to {@link OperatingSystemMXBean}.
- *
- * <p>Prior to JDK 14 {@link OperatingSystemMXBean} did not take into consideration container memory limits when reporting total system
- * memory. Therefore attempts to use this implementation on earlier JDKs will result in an {@link SystemMemoryInfoException}.
  */
-@SuppressForbidden(reason = "Using com.sun internals is the only way to query total system memory")
 public final class DefaultSystemMemoryInfo implements SystemMemoryInfo {
-    private final OperatingSystemMXBean operatingSystemMXBean;
+    private final LongSupplier totalMemory;
 
     public DefaultSystemMemoryInfo() {
-        this.operatingSystemMXBean = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+        this.totalMemory = getOSBeanMemoryGetter();
     }
 
-    @Override
-    @SuppressWarnings("deprecation")
-    public long availableSystemMemory() throws SystemMemoryInfoException {
-        if (JavaVersion.majorVersion(JavaVersion.CURRENT) < 14) {
-            throw new SystemMemoryInfoException("The minimum required Java version is 14 to use " + this.getClass().getName());
-        }
+    @SuppressForbidden(reason = "Using com.sun internals is the only way to query total system memory")
+    private static LongSupplier getOSBeanMemoryGetter() {
+        OperatingSystemMXBean bean = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+        return bean::getTotalMemorySize;
+    }
 
-        return operatingSystemMXBean.getTotalPhysicalMemorySize();
+    @Override
+    public long availableSystemMemory() {
```

Comment:
Since we've removed `SystemMemoryInfoException` from the signature here we should probably remove it everywhere else as well (catch statements, `SystemMemoryInfo` interface, etc). As far as I can tell this was the only place we ever threw such an exception, unless we want to keep this in place for furture implementations?

Reply:
I've removed `SystemMemoryInfoException`